### ISSUE TITLE
SubImage demands 'static

### ISSUE THREAD

**Issue by boomshroom** [image-rs/image#670]
I'm trying to recursively modify an image using sub_image, but any ImageBuffers that I use have to be available statically. The only way I know that I can do that if with a `mut static` which requires unsafe.

I don't think even using a mutex would work because the reference would be captured by a MutexGuard which has a limited lifetime.

My suggestion is to remove the `'static` lifetime restriction and just have the SubImage hold onto the reference until it drops. It's still safe because until the SubImage is dropped, nothing else can borrow the parent because it has an out standing mutable reference.

TLDR: SubImage is currently unusable without unsafe because `'static`.

**Comment by boomshroom** [comment]
After wrapping in an Option because I can't initialize it statically, rustc now claims I have a static destructor and demands that I use nightly and `#![feature(drop_types_in_const)]`. This seems like a lot of work for divide and conquer image editing.

**Comment by dbkaplun** [comment]
Any progress on this?

**Comment by HeroicKatora** [comment]
Should be fixed. The hard bound of `'static` was removed in the `0.21.0` series. Only `ImageBuffer` still requires its subpixels to satisfy that bound.
